packages/nocodb/src/plugins/s3/S3.ts

```diff
@@ -104,6 +106,14 @@ export default class S3 implements IStorageAdapterV2 {
     });
   }
 
+  public async getSignedUrl(key, expires = 7200) {
```

Comment:
better call it `expiresInSeconds`

Reply:
1. Upload part is not changed other than naming, we used to get path from query params now we decide instead.

I am assuming we are running instance with NC_SECURE_ATTACHMENTS set to true.
Read/List apis over attachment columns will return signedPath/signedUrl beside path/url which will be available for set period of time and url's will be not accessible anymore.

The steps you are explaining is pretty much correct.
- We bind path prefixed by set expire time to a signed link (temporary link) within cache
- When user tries to fetch that link we get path using that link.
As api calls trigger signing of the links only valid users can generate those links.

2. True, I will rename them accordingly
3. `http://localhost:8080/dltemp/1696436400000/2023/10/04/hAAj58GvTkWBGYJbpmzlcz0YfLHc43fE/lnSUOhOMuUO2OZuOri.jpg`
In order to avoid brute force attacks I will add a random token there as well which is missing currently
4. Sure, thank you for pointing out

Let me know if you still have any doubts/recommendations